src/cryptonote_core/service_node_list.h

```diff
@@ -158,19 +164,24 @@ namespace service_nodes
     END_SERIALIZE()
   };
 
-  template<typename T>
-  void loki_shuffle(std::vector<T>& a, uint64_t seed)
+
+  template<typename RandomIt>
+  void loki_shuffle(RandomIt begin, RandomIt end, uint64_t seed)
   {
-    if (a.size() <= 1) return;
+    const size_t size = std::distance(begin, end);
```

Comment:
should probably be `const auto size = std::distance(begin, end);` - you're potentially dropping negative values here (if `end < begin`).

Reply:
The mixed signed/unsigned types are a nuissance to deal with later in the function, but I've added a check before the distance call to ensure it will be positive (and in fact, >= 1).